// @google/adk/dist/web/a2a/a2a_event.js
/**
 * @license
 * Copyright 2026 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */
import { randomUUID } from "../utils/env_aware_utils.js";
var MessageRole = /* @__PURE__ */ ((MessageRole2) => {
  MessageRole2["USER"] = "user";
  MessageRole2["AGENT"] = "agent";
  return MessageRole2;
})(MessageRole || {});
function isTaskStatusUpdateEvent(event) {
  return (event == null ? void 0 : event.kind) === "status-update";
}
function isTaskArtifactUpdateEvent(event) {
  return (event == null ? void 0 : event.kind) === "artifact-update";
}
function isMessage(event) {
  return (event == null ? void 0 : event.kind) === "message";
}
function isTask(event) {
  return (event == null ? void 0 : event.kind) === "task";
}
function getEventMetadata(event) {
  if (isTaskArtifactUpdateEvent(event)) {
    return event.artifact.metadata || {};
  }
  if (isTaskStatusUpdateEvent(event) || isTask(event) || isMessage(event)) {
    return event.metadata || {};
  }
  return {};
}
function isFailedTaskStatusUpdateEvent(event) {
  return (isTaskStatusUpdateEvent(event) || isTask(event)) && event.status.state === "failed";
}
function isTerminalTaskStatusUpdateEvent(event) {
  return (isTaskStatusUpdateEvent(event) || isTask(event)) && ["completed", "failed", "canceled", "rejected"].includes(event.status.state);
}
function isInputRequiredTaskStatusUpdateEvent(event) {
  return (isTaskStatusUpdateEvent(event) || isTask(event)) && event.status.state === "input-required";
}
function getFailedTaskStatusUpdateEventError(event) {
  var _a;
  if (!isFailedTaskStatusUpdateEvent(event)) {
    return void 0;
  }
  const parts = ((_a = event.status.message) == null ? void 0 : _a.parts) || [];
  if (parts.length === 0) {
    return void 0;
  }
  if (parts[0].kind !== "text") {
    return void 0;
  }
  return parts[0].text;
}
function createTaskSubmittedEvent({
  taskId,
  contextId,
  message
}) {
  return {
    kind: "status-update",
    taskId,
    contextId,
    final: false,
    status: {
      state: "submitted",
      message,
      timestamp: (/* @__PURE__ */ new Date()).toISOString()
    }
  };
}
function createTask({
  contextId,
  message,
  taskId
}) {
  return {
    kind: "task",
    id: taskId || randomUUID(),
    contextId,
    history: [message],
    status: {
      state: "submitted",
      timestamp: (/* @__PURE__ */ new Date()).toISOString()
    }
  };
}
function createTaskWorkingEvent({
  taskId,
  contextId,
  message
}) {
  return {
    kind: "status-update",
    taskId,
    contextId,
    final: false,
    status: {
      state: "working",
      message,
      timestamp: (/* @__PURE__ */ new Date()).toISOString()
    }
  };
}
function createTaskCompletedEvent({
  taskId,
  contextId,
  metadata = {}
}) {
  return {
    kind: "status-update",
    taskId,
    contextId,
    final: true,
    status: {
      state: "completed",
      timestamp: (/* @__PURE__ */ new Date()).toISOString()
    },
    metadata
  };
}
function createTaskArtifactUpdateEvent({
  taskId,
  contextId,
  artifactId,
  parts = [],
  metadata,
  append,
  lastChunk
}) {
  return {
    kind: "artifact-update",
    taskId,
    contextId,
    append,
    lastChunk,
    artifact: {
      artifactId: artifactId || randomUUID(),
      parts,
      metadata
    }
  };
}
function createTaskFailedEvent({
  taskId,
  contextId,
  error,
  metadata
}) {
  return {
    kind: "status-update",
    taskId,
    contextId,
    status: {
      state: "failed",
      message: {
        kind: "message",
        messageId: randomUUID(),
        role: "agent",
        taskId,
        contextId,
        parts: [
          {
            kind: "text",
            text: error.message
          }
        ]
      },
      timestamp: (/* @__PURE__ */ new Date()).toISOString()
    },
    metadata,
    final: true
  };
}
function createTaskInputRequiredEvent({
  taskId,
  contextId,
  parts
}) {
  return {
    kind: "status-update",
    taskId,
    contextId,
    final: true,
    status: {
      state: "input-required",
      message: {
        kind: "message",
        messageId: randomUUID(),
        role: "agent",
        taskId,
        contextId,
        parts
      },
      timestamp: (/* @__PURE__ */ new Date()).toISOString()
    }
  };
}
function createInputMissingErrorEvent({
  taskId,
  contextId,
  parts
}) {
  return {
    kind: "status-update",
    taskId,
    contextId,
    final: true,
    status: {
      state: "input-required",
      message: {
        kind: "message",
        messageId: randomUUID(),
        role: "agent",
        taskId,
        contextId,
        parts
      },
      timestamp: (/* @__PURE__ */ new Date()).toISOString()
    }
  };
}
export {
  MessageRole,
  createInputMissingErrorEvent,
  createTask,
  createTaskArtifactUpdateEvent,
  createTaskCompletedEvent,
  createTaskFailedEvent,
  createTaskInputRequiredEvent,
  createTaskSubmittedEvent,
  createTaskWorkingEvent,
  getEventMetadata,
  getFailedTaskStatusUpdateEventError,
  isFailedTaskStatusUpdateEvent,
  isInputRequiredTaskStatusUpdateEvent,
  isMessage,
  isTask,
  isTaskArtifactUpdateEvent,
  isTaskStatusUpdateEvent,
  isTerminalTaskStatusUpdateEvent
};
